pkg/objectstate/sched/sched.go

```diff
@@ -0,0 +1,50 @@
+package sched
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nrsv1alpha1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1alpha1"
+)
+
+const DefaultSchedulerPluginImage = "quay.io/openshift-kni/scheduler-plugins:4.10-snapshot"
+
+func UpdateDeploymentImageSettings(dp *appsv1.Deployment, userImageSpec, builtinImageSpec string, builtinPullPolicy corev1.PullPolicy) {
+	// There is only a single container
+	cnt := &dp.Spec.Template.Spec.Containers[0]
+	if userImageSpec != "" {
+		// we don't really know what's out there, so we minimize the changes.
+		cnt.Image = userImageSpec
+		klog.V(3).InfoS("Exporter image", "reason", "user-provided", "pullSpec", userImageSpec)
+		return
+	}
+
+	if builtinImageSpec == "" {
+		cnt.Image = DefaultSchedulerPluginImage
+		klog.V(3).InfoS("Exporter image", "reason", "default", "pullSpec", builtinImageSpec)
+		return
+	}
+
+	cnt.Image = builtinImageSpec
```

Comment:
for the RTE, we need this when we use the operator image as RTE image. For the scheduler we don't need to change the pullPolicy, it's sufficient to set it as we want in the `deployment.yaml`. Please remove.

Reply:
This is not the ```pullPolicy``` but the image. I'll remove the ```pullPolicy```  field

Reply:
please remove the `builtinImageSpec` from this function. We really need to require the user to set a `imageSpec`

Reply:
The builtinImageSpec does exactly that:
https://github.com/Tal-or/numaresources-operator/blob/numares-sched-controller/controllers/numaresourcesscheduler_controller.go#L168

I can change the name to be ```imageSpec``` if that's what you mean

Reply:
we don't want a default here. We need the user to explicitely set it or fail.